test/unit/util_test.js

```diff
@@ -488,6 +488,37 @@ describe('Util', function ()
     }
   });
 
+  it('Util.url.appendRetryParam()', function () {
```

Comment:
let's write this test with separated `it` functions:
```
[
        {
          option: {
            url: 'http://www.something.snowflakecomputing.com',
            retryCount: 3,
            retryReason: 429,
            includeRetryContext: true,
          },
          result: 'http://www.something.snowflakecomputing.com?retryCount=3&retryReason=429'
        },
        {
          option: {
            url: 'http://www.something.snowflakecomputing.com',
            retryCount: 3,
            retryReason: 429,
            includeRetryContext: false,
          },
          result: 'http://www.something.snowflakecomputing.com?retryCount=3'
        }
      ].forEach((...) => {
  it(..)
})
```
and let's it contains more description of the case e.g.
`append retry params with retry reason` and `append retry params without retry reason`

Reply:
Added.